core/vote/vote_signer.go

```diff
@@ -0,0 +1,86 @@
+package vote
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/prysmaticlabs/prysm/crypto/bls"
+	validatorpb "github.com/prysmaticlabs/prysm/proto/prysm/v1alpha1/validator-client"
+	"github.com/prysmaticlabs/prysm/validator/keymanager"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/metrics"
+)
+
+const (
+	voteSignerTimeout = time.Second * 5
+)
+
+type VoteSigner struct {
+	km *keymanager.IKeymanager
+}
+
+func NewVoteSigner(km *keymanager.IKeymanager) (*VoteSigner, error) {
```

Comment:
As the design module, voteManager will be mainly for manage his own votes and check for rules. I think it will be messy and too redundant for voteManager if combine these two module together